lib/SILOptimizer/Mandatory/TFConstExpr.cpp

```diff
@@ -133,16 +129,21 @@ namespace {
 
     /// Invariant: Before the call, `calculatedValues` must not contain `addr`
     /// as a key.
-    SymbolicValue createMemoryObject(SILValue addr, SymbolicValue initialValue){
+    SymbolicValue createMemoryObject(SILValue addr) {
       assert(!calculatedValues.count(addr));
-      unsigned objectID = memoryObjects.size();
-      auto objectType = simplifyType(addr->getType().getASTType());
-      memoryObjects.push_back({initialValue, objectType});
-
-      auto result = SymbolicValue::getAddress(objectID);
+      auto type = simplifyType(addr->getType().getASTType());
+      auto *memObject =
+        SymbolicValueMemoryObject::create(type, evaluator.getAllocator());
+      auto result = SymbolicValue::getAddress(memObject);
       setValue(addr, result);
       return result;
     }
+    SymbolicValue createMemoryObject(SILValue addr, SymbolicValue initialValue){
```

Comment:
similar to above: can we unify these two versions of `createMemoryObject()`, so that we don't set the value of a memory object multiple times when creating/initializing one?